programs/ziti-edge-tunnel/windows-service.c

```diff
@@ -472,3 +472,42 @@ DWORD LphandlerFunctionEx(
 
     return 0;
 }
+
+
+BOOL SetPrivilege(HANDLE hToken, LPCTSTR lpszPrivilege, BOOL bEnablePrivilege)
```

Comment:
if anything goes wrong in this function it's going to be really hard to tall what happened from the logs

Reply:
What would you like to see happen? I'm not sure there's any 'good' way to handle that situation

Reply:
I didn't look into the win32 api to see if the failure cases can be "handled". I'm assuming they can't. I was thinking more about if there's any value to a support engineer or developer in knowing whether LookupPrivilegeValue failed vs AdjustTokenPrivileges? Without knowing the API and reasons for failures it seems like it would be useful to log errors when a function fails, possibly with an error code if available.

Reply:
Yeah. This runs as basically 'the first command' so logging isn't initialized yet. I suppose it might make sense to move where this happens until after the logging is initialized so that it can be tested and logged - I'll do that...